tests/test_dict_plotting.py

```diff
@@ -43,8 +43,8 @@ def test_plot_dictionary(self):
     print(data)
     
     ploter = plot_dictionary()
-
-    plot_string = "-dn density -y mat1 -x time -sa test.png --hide_plot"
+    tmp_dir = tempfile.TemporaryDirectory()
+    plot_string = "-dn density -y mat1 -x time -sa "+tmp_dir.name+"test.png --hide_plot"
```

Comment:
Here and elsewhere, this will directly concatenate the path to the filename, so you end up with i.e., `/tmp/tmp1mxbcs_mtest.png`. The test still passes/works, but I suspect you meant to produce i.e., `/tmp/tmp1mxbcs_m/test.png`.

Maybe consider constructing the output file paths like this:

```diff
--- a/tests/test_dict_plotting.py
+++ b/tests/test_dict_plotting.py
@@ -44,7 +44,8 @@ class test_dict_plotting(unittest.TestCase):
     
     ploter = plot_dictionary()
     tmp_dir = tempfile.TemporaryDirectory()
-    plot_string = "-dn density -y mat1 -x time -sa "+tmp_dir.name+"test.png --hide_plot"
+    outfile = os.path.join(tmp_dir.name, "test.png")
+    plot_string = f"-dn density -y mat1 -x time -sa {outfile} --hide_plot
```

Reply:
I was wondering about this. I'll modify to match your suggestion.